In [5]:
#Import packages Numpy and Pandas

import pandas as pd
import numpy as np
pd.options.display.max_columns = None
pd.options.display.max_rows = None

In [6]:
#Read in dataset obtained from www.kaggle.com/datasets/kartik2112/fraud-detection/ that is currently in working directory

df_train = pd.read_csv('fraudTrain.csv')

In [7]:
#Check the head and shape of our dataset

print(df_train.head())

df_train.shape

   Unnamed: 0 trans_date_trans_time            cc_num  \
0           0   2019-01-01 00:00:18  2703186189652095   
1           1   2019-01-01 00:00:44      630423337322   
2           2   2019-01-01 00:00:51    38859492057661   
3           3   2019-01-01 00:01:16  3534093764340240   
4           4   2019-01-01 00:03:06   375534208663984   

                             merchant       category     amt      first  \
0          fraud_Rippin, Kub and Mann       misc_net    4.97   Jennifer   
1     fraud_Heller, Gutmann and Zieme    grocery_pos  107.23  Stephanie   
2                fraud_Lind-Buckridge  entertainment  220.11     Edward   
3  fraud_Kutch, Hermiston and Farrell  gas_transport   45.00     Jeremy   
4                 fraud_Keeling-Crist       misc_pos   41.96      Tyler   

      last gender                        street            city state    zip  \
0    Banks      F                561 Perry Cove  Moravian Falls    NC  28654   
1     Gill      F  43039 Riley Greens Suite 39

(1296675, 23)

In [8]:
#Remove Unnamed: 0 column which seems to be a manual index count and is therefore not needed
df_train = df_train.drop('Unnamed: 0', axis = 1)

print(df_train.head())

df_train.shape

  trans_date_trans_time            cc_num                            merchant  \
0   2019-01-01 00:00:18  2703186189652095          fraud_Rippin, Kub and Mann   
1   2019-01-01 00:00:44      630423337322     fraud_Heller, Gutmann and Zieme   
2   2019-01-01 00:00:51    38859492057661                fraud_Lind-Buckridge   
3   2019-01-01 00:01:16  3534093764340240  fraud_Kutch, Hermiston and Farrell   
4   2019-01-01 00:03:06   375534208663984                 fraud_Keeling-Crist   

        category     amt      first     last gender  \
0       misc_net    4.97   Jennifer    Banks      F   
1    grocery_pos  107.23  Stephanie     Gill      F   
2  entertainment  220.11     Edward  Sanchez      M   
3  gas_transport   45.00     Jeremy    White      M   
4       misc_pos   41.96      Tyler   Garcia      M   

                         street            city state    zip      lat  \
0                561 Perry Cove  Moravian Falls    NC  28654  36.0788   
1  43039 Riley Greens Suite 393     

(1296675, 22)

In [13]:
#Check for missing or Null values and drop any duplicates that may be present
df_train.isnull().sum()

trans_date_trans_time    0
cc_num                   0
merchant                 0
category                 0
amt                      0
first                    0
last                     0
gender                   0
street                   0
city                     0
state                    0
zip                      0
lat                      0
long                     0
city_pop                 0
job                      0
dob                      0
trans_num                0
unix_time                0
merch_lat                0
merch_long               0
is_fraud                 0
dtype: int64

In [14]:
print(f"Amount of duplicated rows removed: {df_train.duplicated().sum()}")

df_train = df_train.drop_duplicates()

Amount of duplicated rows removed: 0


From this we can see that our dataset is a set of 1,296,675 transactions with 23 features to the transaction. Our target variable is the column "is_fraud" with the rest of the 22 columns being our feature variables such as time of transaction, amount of transaction, spending category, and merchant category. Some personal imformation about the card holders such as Name my not be relavent to our fraud model but those can be dropped later in our Feature Engineering. As we can see, we have no missing values in any of our columns nor do we have any duplicates. This tracks for what is expected for a record of well kept transactional data